test/IntegrationTests/Helpers/OtlpResourceExpector.cs

```diff
@@ -67,6 +74,30 @@ public void AssertExpectations(TimeSpan? timeout = null)
         }
     }
 
+    private static void CheckExistence(List<string> existenceChecks, RepeatedField<KeyValue>? actualResourceAttributes)
```

Comment:
Can it be part of AssertExpectations?
It will be less error-prone as you do not need to call seconds method to Assert everything.